### ISSUE TITLE
Adapter updates for Javascript-Adapter should restart Instances sequentially 

### ISSUE THREAD

**Issue by seb2010** [ioBroker/ioBroker#512]
I have a scenario with multiple Javascript instances. The instances do require multiple libraries to be installed in their node-subfolder. When I now update the overall adapter to a higher version, once the instances get restarted, they all start at once. I frequently get log-error having problems with the additional node-libraries of the instances.
I am suspecting it to be because each instance is firing the "npm install" command for the libraries at the same time and maybe this leads to hickups. To bypass this, the restart sequence of the instances of an adapter should be sequential with a defined delay of some seconds.

**Comment by Apollon77** [comment]
I would declatre this as duplicate of https://github.com/ioBroker/ioBroker.js-controller/issues/2006 ... when we change the way we handle npm installs we need also prevent concurrent runs

**Comment by mcm1957** [comment]
@Apollon77 
But independent of npm I would consider it a good idea to start instances of ONE Adapter sequently one after the other.

The instances of one identical adapter could likly access the same cloudservice even with different parameters but of course from same ip. This could cause troubles at some cloud services.

I suggest to aevaluate this.

**Comment by Apollon77** [comment]
Then this would be another request for js.-controller ... I thiought this is already done ... mabye we need to add a delay

**Comment by seb2010** [comment]
@Apollon77 : a delay itself might not cut it. I have 5+ libraries to load and even on a faster i7 NUC the installation of these packages takes 20+ seconds. The start of the next instance should be bound to the state of the previous instance where the javascript controller really started (all sub-script due to start were stared) and subscribed states (emitting "received all states") etc, as the installation process of the libraries need to finish first for sure. Maybe this state corresponds to a variable under system.*

**Comment by Apollon77** [comment]
In fcat it also touches https://github.com/ioBroker/ioBroker.js-controller/issues/2006 as idea to optimize the whole "custom npom installing" ...
